zipline/pipeline/factors/factor.py

```diff
@@ -621,6 +627,120 @@ def rank(self, method='ordinal', ascending=True, mask=NotSpecified):
         """
         return Rank(self, method=method, ascending=ascending, mask=mask)
 
+    @expect_types(
+        target_slice=Slice,
+        correlation_length=int,
+        mask=(Filter, NotSpecifiedType),
+    )
+    def rolling_pearsonr(self,
+                         target_slice,
+                         correlation_length,
+                         mask=NotSpecified):
+        """
+        Construct a new Factor that computes rolling pearson correlation
+        coefficients between the columns of `self` and the target slice.
+
+        Parameters
+        ----------
+        target_slice : zipline.pipeline.slice.Slice
+            The column of data with which to compute correlations against each
+            column of data produced by `self`.
+        correlation_length : int
+            Length of the lookback window over which to compute each
+            correlation coefficient.
+        mask : zipline.pipeline.Filter, optional
+            A Filter describing which assets should have their correlation with
+            the target slice computed each day.
+
+        Returns
+        -------
+        correlations : zipline.pipeline.factors.RollingPearson
+            A new Factor that will compute correlations between `target_slice`
+            and the columns of `self`.
+        """
+        return RollingPearson(
```

Comment:
Can we make assertions about the dtypes of `self` and `target_slice` here?  (Or are those assertions made in `RollingPearson`?)